lint/rule_panel_units.go

```diff
@@ -0,0 +1,100 @@
+package lint
+
+import "fmt"
+
+func NewPanelUnitsRule() *PanelRuleFunc {
+	validUnits := []string{
+		// Enumerated from: https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/valueFormats/categories.ts
+
+		// Misc
+		"none", "string",
```

Comment:
Should we leave none as an allowed unit to be set if no other unit makes sense?

Or remove this and let the user add a lint exclusion in that case?

Reply:
I wasn't sure the behavior for a default dashboard/panel, so I just tested it.

When a new panel is created, with no config, no unit is set. Literally the `unit` property is absent in `fieldConfig`.

In the dropdown/picker for units, `none` does not appear to be an option. If you type it in manually, you can set it as a "custom" unit.

For this reason, I would say this shouldn't be an allowed valid unit, and if users choose to explicitly specify "none" they should also create a lint exclusion rule.